src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +58,10 @@ private void Initialize(IEdmEntityType entityType)
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+                
+        /// <summary>
+        /// Get the Change Object collection
+        /// </summary>
+        public virtual IEnumerable ChangedObjectCollection { get; }        
```

Comment:
this is a bit strange; are you expecting folks to do foreach(var o in changedObjectCollection.ChangedObjectCollection) ?  We should remove this if we can, and just make sure the underlying EdmChangedObjectCollection is kept in sync with the EdmChangedObjectCollection<T> OR we should have a protected virtual overridden in the EdmChangedObjectCollection<T> to return the enumerable, and use that in GetEnumerator().